projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -112,6 +112,14 @@ public function admin_init() {
 		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
 	}
 
+	/**
+	 * Check if we have a paid Jetpack Social plan.
+	 */
+	public function has_paid_plan() {
+		$refresh_from_wpcom = true;
```

Comment:
Could we change this to:

```suggestion
	public function has_paid_plan( $refresh_from_wpcom = false ) {
```

And then call it with `true` in `initial_state`? That would stop the call to `has_paid_plan` in `set_up_sharing_limits` from hitting WPCOM for every admin request.

Reply:
As we're going to move this code into the package, we'll merge this as it is, and refactor it in a follow-up PR.